.github/scripts/check-folders.js

```diff
@@ -83,17 +84,27 @@ async function main() {
       const job_optional = optionalTestYaml['jobs'][key]
       const matrix_optional =
         Boolean(job_optional) && Boolean(job_optional.strategy) ? job_optional.strategy.matrix : {}
+
       const folders_optional = Object.keys(matrix_optional)
         .filter((key) => {
+          // These are in optional-test.yaml, under the `platforms` directory
           return !keysToIgnore.includes(key)
         })
-        .reduce((acc, key) => acc.concat(...matrix_optional[key]), [])
+        .reduce((acc, key) => {
+          return acc.concat(...matrix_optional[key])
+        }, [])
+
+      // These are in optional-test.yaml, under the `platforms` directory
+      // But the names don't match the directory name
+      // Example `platforms-m1-office` is in `platforms` directory
+      // So this renames the key so it's correctly processed
+      if (key === 'platforms-m1-office' || key === 'platforms-codesandbox') {
+        key = 'platforms'
+      }
```

Comment:
Current hack to make the checks-folders pass without introducing a new top-level directory

I suggest we keep the hack for now and merge, and I'm open to suggestion to clean this up in a follow-up PR.